### PR TITLE
Feature/log to cloudwatch

### LINKED ISSUE #24: Stream ec2 instances logging to cloudwatch
Stream ec2 instance logging to cloudwatch
- /var/log/user_data
- security related logs
- runner log

Make retention time configureable.

### PR DESCRIPTION
**Pull request by npalm**
Sent instance log of the agent process, installation (user_data) and system mesages to cloudwatch. Fix #24

### CODE REVIEW

**Comment by npalm** [comment]
image

File: modules/runners/variables.tf [hunk 1]
```diff
@@ -244,3 +244,37 @@ variable "runner_iam_role_managed_policy_arns" {
...
+}
+
+variable "cloudwatch_config" {
+  description = "(optional) to replace the module default cloudwatch log config. See https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-Agent-Configuration-File-Details.html for details."
+  type        = string
+  default     = null
+}
+
+variable "runner_log_files" {
+  description = "(optional) List of logfiles to sent to cloudwatch."
```
Comment (gertjanmaas) [inline modules/runners/variables.tf:261]:
```suggestion
  description = "(optional) List of logfiles to send to cloudwatch."
```

File: variables.tf [hunk 2]
```diff
@@ -262,3 +262,37 @@ variable "runner_iam_role_managed_policy_arns" {
   type        = list(string)
   default     = []
 }
+
+variable "enable_cloudwatch_agent" {
+  description = "Enabling the cloudwatch agent on the ec2 runner instances, the runner contains default config. Configuration can be overrided via `cloudwatch_config`."
```
Comment (gertjanmaas) [inline variables.tf:267]:
```suggestion
  description = "Enabling the cloudwatch agent on the ec2 runner instances, the runner contains default config. Configuration can be overridden via `cloudwatch_config`."
```

File: variables.tf [hunk 3]
```diff
@@ -262,3 +262,37 @@ variable "runner_iam_role_managed_policy_arns" {
...
+  type        = bool
+  default     = true
+}
+
+variable "cloudwatch_config" {
+  description = "(optional) to replace the module default cloudwatch log config. See https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-Agent-Configuration-File-Details.html for details."
```
Comment (gertjanmaas) [inline variables.tf:273]:
```suggestion
  description = "(optional) Replaces the module default cloudwatch log config. See https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-Agent-Configuration-File-Details.html for details."
```

File: variables.tf [hunk 4]
```diff
@@ -262,3 +262,37 @@ variable "runner_iam_role_managed_policy_arns" {
...
+  type        = string
+  default     = null
+}
+
+variable "runner_log_files" {
+  description = "(optional) List of logfiles to sent to cloudwatch."
```
Comment (gertjanmaas) [inline variables.tf:279]:
```suggestion
  description = "(optional) List of logfiles to send to cloudwatch."
```

**Review by gertjanmaas — CHANGES_REQUESTED** [review]
LGTM, only fixed some small typos

**Comment by npalm** [comment]
@gertjanmaas PR is finished, docs updated and tested

**Review by gertjanmaas — APPROVED** [review]

**Pull request merged**